### PR TITLE
feat(messaging): @mention with deep-link to node detail (#6098)

### LINKED ISSUE #6098: Add @mention with deep-link to node detail
## What

Per [meshtastic/design#21](https://github.com/meshtastic/design/issues/21), add @mention support to messaging: typing `@` in the compose field should offer an autocomplete of nearby/known nodes (matched by long name, short name, or user ID), and a received message containing a mention should render it as tappable text that deep-links to that node's detail screen.

No implementation exists on any platform today — this is a from-scratch feature, not a fix.

## Wire format (must match iOS/Web exactly — this is the cross-platform contract)

The message text payload packages a mention as the literal string `@!<hex-id>` (e.g. `@!ffccee11`) — the node's numeric ID, hex-encoded, **not** the display name. This is required so the mention stays valid and correctly resolvable even if the mentioned node's long/short name changes later, and so all three clients parse the exact same wire format regardless of which one composed the message.

## Compose side

- In the message compose input (`feature/messaging/src/commonMain/kotlin/org/meshtastic/feature/messaging/Message.kt`), detect `@` typed in the text field and show an autocomplete list of known nodes, filterable by long name, short name, or user ID.
- On selection, insert `@!<hex-id>` into the underlying text that gets sent (not the display name) — but the compose field should still *show* the friendly name to the user while typing, with the hex form being what actually goes out over the wire via `SendMessageUseCase`.

## Render side

- Wherever message bubbles render text (`feature/messaging/.../component/MessageItem.kt`), detect `@!<hex-id>` tokens and render each as tappable text showing the **current** display name resolved live from the node repository — a name change should update the mention's displayed text without needing to re-send anything; only the hex ID is stable/stored.
- Tapping a mention should deep-link to that node's detail screen, consistent with existing node-tap navigation elsewhere in the app.

## Notification behavior

When the local node's own ID is mentioned in a channel broadcast, that should be treated at least as prominently as a direct message for notification purposes — e.g. still notify even if channel notifications are otherwise muted, since a mention is a targeted ping within an otherwise-broadcast conversation. Coordinate the exact behavior with the iOS/Web implementations so all three treat a self-mention consistently.

## Reference

Cross-platform alignment tracking issue: meshtastic/design#21.

### PR DESCRIPTION
**Pull request by jamesarich**
## Why

Closes #6098. A mention is a *targeted* ping inside an otherwise-broadcast channel — but today there's no way to point at a specific node in a message, and no way for the pinged user to notice or jump to who was mentioned. This adds @mentions end-to-end, following the cross-platform wire contract in meshtastic/design#21.

**Wire format (the cross-platform contract):** a mention is stored/sent as the literal token `@!<hex>` — i.e. `@` + the node's `User.Id` (`@!ffccee11`), the numeric id hex-encoded, **never** the display name. This keeps the mention valid when a node is renamed, and makes Android/iOS/Web parse identical bytes. Verified against design#21 (which specifies `"@!ffccee11"` verbatim); no `@!` mention code exists on iOS/Web yet, so this follows the agreed spec rather than mirroring existing client code.

## 🌟 What's new

- **Compose autocomplete** — typing `@` shows a filterable list of known nodes (match on long name, short name, or user id). Selecting one inserts the wire token `@!<hex>`, while a Compose `OutputTransformation` displays the friendly name as you type. The buffer holds the hex, so `SendMessageUseCase` ships the wire form unchanged.
- **Tappable, live-resolved mentions** — received `@!<hex>` tokens render as tappable text showing the node's **current** display name, resolved live from the node map (a rename reflows the label with no re-send). Tapping deep-links to that node's detail screen, reusing the existing chip→node-detail navigation.
- **Self-mention notifications** — when the local node's own id is mentioned, the message notifies even if the channel/contact is muted (forces `isSilent = false`). Kept at normal broadcast importance for now; can be escalated to DM-priority later if the other clients land that way.

## 🛠️ How

- `core/ui` `AutoLinkText`: two new **optional** params (`mentionName`, `onMentionClick`, default `null` → existing callers unchanged). Mention detection slots in alongside the existing URL/email/phone linkifier; substitution runs first so all offsets line up.
- `feature/messaging` `Message.kt`: `@`-query detection over `TextFieldState` text+selection, a suggestion popup above the input, and the `OutputTransformation` for display.
- `MessageListPaged` / `MessageItem`: resolve `!<hex>` → `Node` via the existing `nodeMap`, wire the resolver + tap through to `AutoLinkText`.
- `core/data` `MeshDataHandlerImpl.handlePacketNotification`: mute is overridden when the text contains `@` + `getMyId()`.

## 🧹 Notes / deferred

- Mentions render in the normal message bubble; the search-highlight view and the tiny quoted-reply snippet still show raw `@!<hex>` (deliberately out of scope to keep the diff tight — easy follow-up).
- `OutputTransformation` substitutes display back-to-front; editing *inside* a substituted token is imperfect, but the caret sits outside tokens in normal use.

## Testing performed

- New unit test `MentionQueryTest` covers the `@`-query parser edge cases (start-of-text, after-whitespace, bare `@`, mid-word/email rejection, completed-token no-retrigger, non-collapsed selection).
- Full baseline green: `spotlessCheck`, `detekt`, `assembleDebug`, `test`, `allTests`, `kmpSmokeCompile`.

## Summary by CodeRabbit

* **New Features**
  * Added `@mention` autocomplete to the message composer with suggestions and token insertion.
  * Mentions in chat now render as tappable friendly names, alongside existing linkification.
* **Bug Fixes**
  * Mentioned messages can trigger notifications even when chat/sender is muted (when the text mentions the local node).
* **Tests**
  * Added unit tests for mention query detection and mention-token/regex matching.

### CODE REVIEW

File: core/data/src/commonMain/kotlin/org/meshtastic/core/data/manager/MeshDataHandlerImpl.kt [hunk 1]
```diff
@@ -414,7 +414,13 @@ class MeshDataHandlerImpl(
...
         val conversationMuted = packetRepository.value.getContactSettings(contactKey).isMuted
         val nodeMuted = nodeManager.getNodeById(dataPacket.from.orEmpty())?.isMuted == true
-        val isSilent = conversationMuted || nodeMuted
+        // A mention of our own id is a targeted ping, so it should still notify even in a muted channel/contact.
+        val myId = nodeManager.getMyId()
+        val mentionsMe =
+            dataPacket.dataType == PortNum.TEXT_MESSAGE_APP.value &&
+                myId.length > 1 &&
+                dataPacket.text?.contains("@$myId", ignoreCase = true) == true
+        val isSilent = (conversationMuted || nodeMuted) && !mentionsMe
```
[the remark this answers is not in the document]
Comment (jamesarich) [reply core/data/src/commonMain/kotlin/org/meshtastic/core/data/manager/MeshDataHandlerImpl.kt:422]: Fixed. Replaced the loose `contains("@\$myId")` with a boundary-safe check using the same `(?![0-9a-fA-F])` lookahead as the rest of the mention stack. Also centralized that logic into `core/model` (`textMentionsNode` + `MENTION_TOKEN_REGEX`) so all three surfaces share one source of truth, and added `MentionTest` covering the `@!ffccee11a` prefix case.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: feature/messaging/src/commonMain/kotlin/org/meshtastic/feature/messaging/Message.kt [hunk 2]
```diff
@@ -523,55 +560,115 @@ private fun MessageInput(
...
+            // Direct byte limiting via inputTransformation in TextFieldState is complex.
+            // The current approach (show error, disable send) is generally preferred for UX.
+            // If strict real-time byte trimming is required, it needs careful handling of
+            // cursor position and multi-byte characters, likely outside simple inputTransformation.
+            trailingIcon = {
+                IconButton(onClick = { if (canSend) onSendMessage() }, enabled = canSend) {
+                    Icon(imageVector = MeshtasticIcons.Send, contentDescription = stringResource(Res.string.send))
+                }
+            },
+        )
```
[the remark this answers is not in the document]
Comment (jamesarich) [reply feature/messaging/src/commonMain/kotlin/org/meshtastic/feature/messaging/Message.kt:661]: Fixed. While the mention popup is open, Enter / IME send / tapping the send icon now complete the top suggestion (via a shared `onSendAction` + `insertMention`) instead of sending the raw incomplete `@query`; normal send behavior is unchanged when no suggestion is active.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: core/model/src/commonMain/kotlin/org/meshtastic/core/model/Mention.kt [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+val MENTION_TOKEN_REGEX = Regex("""@(![0-9a-fA-F]{8})(?![0-9a-fA-F])""")
+
+/**
+ * True if [text] @mentions the node whose hex user id is [userId] (e.g. `!ffccee11`), honoring the token boundary — a
+ * trailing hex digit (`@!ffccee11a`) is a different, longer id and is not a match.
+ */
+fun textMentionsNode(text: String?, userId: String): Boolean {
+    if (text.isNullOrEmpty() || userId.length <= 1) return false
+    return Regex("@${Regex.escape(userId)}(?![0-9a-fA-F])", RegexOption.IGNORE_CASE).containsMatchIn(text)
+}
```
[the remark this answers is not in the document]
Comment (jamesarich) [reply core/model/src/commonMain/kotlin/org/meshtastic/core/model/Mention.kt:35]: Fixed. `textMentionsNode` now derives from the shared `MENTION_TOKEN_REGEX` — it scans for tokens and compares the captured id to `userId` case-insensitively — so the format/boundary rule lives in exactly one place and no new `Regex` is compiled per call. Also added the case-insensitivity test flagged in the same review.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
